src/ucp/tag/offload.c

```diff
@@ -262,78 +387,23 @@ int ucp_tag_offload_post(ucp_request_t *req, ucp_request_queue_t *req_queue)
             UCP_WORKER_STAT_TAG_OFFLOAD(worker, BLOCK_SW_PEND);
             return 0;
         }
-    } else if (worker->tm.expected.wildcard.sw_count ||
-               req_queue->sw_count) {
-        /* There are some requests which must be completed in SW */
-        UCP_WORKER_STAT_TAG_OFFLOAD(worker, BLOCK_SW_PEND);
-        return 0;
+    } else if (req_queue->sw_count || worker->tm.expected.wildcard.sw_count) {
```

Comment:
worker->tm.expected.wildcard.sw_count is checked twice. can we avoid that? maybe check it 1st in a separate if..else case?